devops/nightly.yml

```diff
@@ -13,182 +19,275 @@ schedules:
+      matrix:
+        GCC Debug:
+          CC: gcc
+          CXX: g++
+          CXXFLAGS:
+          BuildType: Debug
+          Asan: Off
+        GCC Release:
+          CC: gcc
+          CXX: g++
+          CXXFLAGS:
+          Asan: Off
+          BuildType: Release
+        Clang Debug:
+          CC: clang
+          CXX: clang++
+          CXXFLAGS: -stdlib=libstdc++
+          BuildType: Debug
+          Asan: Off
+        Clang Release:
+          CC: clang
+          CXX: clang++
+          CXXFLAGS: -stdlib=libstdc++
+          Asan: Off
+          BuildType: Release
+        Clang Debug (ASAN):
+          CC: clang
+          CXX: clang++
+          CXXFLAGS: -stdlib=libstdc++
+          BuildType: Debug
+          Asan: On
+        Clang Release (ASAN):
+          CC: clang
+          CXX: clang++
+          CXXFLAGS: -stdlib=libstdc++
+          BuildType: Release
+          Asan: On
+    steps:
+    - checkout: self
+
+    - script: |
+        set -eo pipefail
+        git submodule init
+        git submodule update --depth 1 --recursive
+      displayName: 'Checkout submodules'
+
+    - template: ./steps/linux/dependencies.yml
+
+    - task: CMake@1
+      displayName: 'CMake'
+      inputs:
+        cmakeArgs: |
+          .. -GNinja -DCMAKE_BUILD_TYPE=$(BuildType) -DCMAKE_CXX_FLAGS=$(CXXFLAGS) -DENABLE_ASSERTS=ON -DUSE_ASAN=$(Asan) -DVERONA_CI_BUILD=On -DRT_TESTS=ON
+
+    - script: |
+        set -eo pipefail
+        ninja
+      workingDirectory: build
+      displayName: 'Compile'
+
+    - template: ./steps/linux/test.yml
+      parameters:
+        WorkingDirectory: build
+
+- stage: LinuxFull
+  displayName: Linux (Verona + LLVM)
+  dependsOn: []
+  jobs:
+  - job:
+    displayName: Linux
+    pool:
+      vmImage: 'ubuntu-20.04'
+    timeoutInMinutes: 120
+    strategy:
+      matrix:
+        GCC Release:
+          CC: gcc
+          CXX: g++
+          CXXFLAGS:
+          Asan: Off
+          BuildType: Release
+        Clang Debug (ASAN):
+          CC: clang
+          CXX: clang++
+          CXXFLAGS: -stdlib=libstdc++
+          BuildType: Debug
+          Asan: On
+
+    variables:
+      - template: ./vars/anybuild-vars.yml
+      - template: ./vars/llvm-vars.yml
+
+    steps:
+    - checkout: self
+
+    - script: |
+        set -eo pipefail
+        git submodule init
+        git submodule update --depth 1 --recursive
+      displayName: 'Checkout submodules'
+
+    - template: ./steps/linux/dependencies.yml
+
+    - bash: |
+        set -euo pipefail
+
+        # drop an AnyBuild.json config file and use it to
+        #   - ignore the CMAKE_BUILD_PARALLEL_LEVEL variable
+        #   - limit the number of concurrent cache lookups
+        echo '{"ActionCache": { "MaxParallelLookups": 10, "IgnoredEnvVars": [ "CMAKE_BUILD_PARALLEL_LEVEL" ] }}' | tee AnyBuild.json
+
+        # remove '-march=native' which is incompatible with remote execution
+        sed -i 's/ -march=native//g' src/rt/CMakeLists.txt
```

Comment:
This is more interesting.  Clearly, `-march=native` is incompatible with remote execution.  By default, AnyBuild ignores all compilation processes that include this switch and forces them to execute locally.  The problem with that is that processes that are executed locally (outside of AnyBuild) cannot be cached, and in this particular case that really hurts the end-to-end build time.

So you can decide whether you want to remove this switch from your `CMakeLists.txt` file(s) or we can remove them here (just for the sake of being able to execute everything with AnyBuild).

Reply:
IIRC, we use this because the default option (minimum common denominator) is really bad for performance. We discussed using a minimal set but that'd break builds on users' older machines. We don't really need anything special.

I guess in CI we don't really need such performance and we could drop the `march` entirely. @mjp41?

Reply:
We can drop it.  This is from quite early in our thinking, and was inherited from snmalloc, but now we don't have that on snmalloc.  I am happy for that to go.

Reply:
Remove `-march=native` from the `CmakeLists.txt` file.